File: ocaml/src/four.ml
open OUnit2

let rec length = function | [] -> 0 | _::rest -> 1 + (length rest)

let test_case =
  "length" >::
    (fun _  ->
       assert_equal 3 (length ["a"; "b"; "c"]); assert_equal 0 (length []))

File: fsharp/Four.fs
module ``NinetyNine Problems - Problem 4``

open NUnit.Framework
open FsUnit

let rec length = function
  | _ :: rest -> 1 + length rest
  | [] -> 0

[<Test>]
let ``Given a list it should return its length`` () =
  length [ "a" ; "b"; "c"; "d"; "e" ] |> should equal 5
